### DISCUSSION TITLE
Save confirmation when pressing a "x" icon

### DISCUSSION THREAD

**Discussion by Cants820** [react-hook-form/react-hook-form#8387] [Q&A] [answered]
Hello, 

Not sure if it's possbile currently im trying to a save confirmation box when you press a "x" button. Basically if you havent saved your form a confirmation pop up would show up saying "You have unsaved changes on this page. Do you want to save?" since not sure where to state here ?
How would I go about in doing this not sure where to start ? 

Thank you,

**Comment by Moshyfawn** [comment, marked as answer]
It's mostly a question around your UI and not RHF, as RHF is only a headless form data manager.

It's up to you to decided what is "not saved form" means to your user. as well, as closing your form modal is managed by your UI, not the form.

To get the current form values to save somewhere, you can call the `getValues` form method in whatever `onClose` callback you have when clicking the close form modal button.

**Comment by Cants820** [comment]
lastly what is a headless form data manager ?

**Reply by bluebill1049** [reply]
headless means there is no UI output, it acting as a skeleton for your component.
